### ISSUE TITLE
Improve Javadoc of ConfigurableFilePermissions.html#unix(int)

### ISSUE THREAD

**Issue by sdavids** [gradle/gradle#30369]
### Expected Behavior

Prevent misuse of API.

It is easy to think that `unix(755)` is `rwxr-xr-x`.

--- 

https://docs.gradle.org/current/javadoc/org/gradle/api/file/ConfigurableFilePermissions.html#unix(int)

> Sets Unix style numeric permissions. See [unix(String)](https://docs.gradle.org/current/javadoc/org/gradle/api/file/ConfigurableFilePermissions.html#unix(java.lang.String)) for details.

https://docs.gradle.org/current/javadoc/org/gradle/api/file/ConfigurableFilePermissions.html#unix(java.lang.String)

> The NUMERIC notation consist of 3 digits having values from 0 to 7. 1st digit represents the OWNER, 2nd represents the GROUP while the 3rd represents OTHER users.

> <p style="font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none; caret-color: rgb(230, 231, 239); color: rgb(230, 231, 239); font-family: Lato, Georgia, &quot;Times New Roman&quot;, Times, serif; font-size: 14px; text-align: left;">Examples:</p>
Numeric | Symbolic | Meaning
-- | -- | --
000 | --------- | no permissions
700 | rwx------ | read, write & execute only for owner

The description and the table suggest using a 3-digit number.

### Current Behavior

https://github.com/gradle/gradle/blob/035834c2fa7b88422635cde176fe9c7a6f6b8bc7/platforms/core-configuration/file-collections/src/main/java/org/gradle/api/internal/file/DefaultConfigurableFilePermissions.java#L96-L100

https://github.com/gradle/gradle/blob/035834c2fa7b88422635cde176fe9c7a6f6b8bc7/platforms/core-configuration/file-collections/src/main/java/org/gradle/api/internal/file/AbstractFilePermissions.java#L32-L35

https://github.com/gradle/gradle/blob/035834c2fa7b88422635cde176fe9c7a6f6b8bc7/platforms/core-configuration/file-collections/src/main/java/org/gradle/api/internal/file/DefaultConfigurableUserClassFilePermissions.java#L38-L42

https://github.com/gradle/gradle/blob/035834c2fa7b88422635cde176fe9c7a6f6b8bc7/platforms/core-configuration/file-collections/src/main/java/org/gradle/api/internal/file/AbstractUserClassFilePermissions.java#L33-L35

```shell
jshell --feedback=silent - <<< 'System.out.println((((755 & 0_700) >> 6) & 4) >> 2 == 1)'
false
 ~ % jshell --feedback=silent - <<< 'System.out.println((((0755 & 0_700) >> 6) & 4) >> 2 == 1)'
true
```

### Context

https://docs.gradle.org/current/userguide/working_with_files.html#sec:setting_file_permissions

> file: read & write for owner, read for group, read for other (0644, rw-r—​r--)
>
> directory: read, write & execute for owner, read & execute for group, read & execute for other (0755, rwxr-xr-x)

https://docs.gradle.org/current/javadoc/org/gradle/api/file/FilePermissions.html

> FILE: read & write for OWNER, read for GROUP, read for OTHER (0644, rw-r--r--)
> DIRECTORY: read, write & execute for OWNER, read & execute for GROUP, read & execute for OTHER (0755, rwxr-xr-x)

Both are OK but could be improved with a `WARNING` that if using a number it should be in octal format.

---

```java
unix(755) // not what one expects
```

```java
unix(0755)
unix("755")
unix("rwxr-xr-x")
```

**Comment by sdavids** [comment]
https://kotlinlang.org/docs/numbers.html#literal-constants-for-numbers

> Octal literals are not supported in Kotlin.

**Comment by sdavids** [comment]
Maybe deprecate `unix(int)`?

**Comment by ljacomet** [comment]
This issue needs a decision from the team responsible for that area. They have been informed. Response time may vary.

***

At the minimum, the introduced `unix(int)` should have javadoc calling out this behaviour.

But we should consider if that method is really needed.

**Comment by jbartok** [comment]
The change is not release notes worthy (in my opinion).

**Comment by jbartok** [comment]
A couple of comments about the PR based on which this issue has been closed. 

First of all, the problem/concern is not entirely realistic. Anybody who has to deal with UNIX style file permissions should be aware, to some extent at least, of the fact that the int value only makes sense in octal. So we have decided not to completely remove the method.

But, there is indeed some room for improvement, and that's what we did:
- we added validation which should catch many of the completely nonsensical inputs, that can be accidentally used
- we improved the JavaDoc to make the meaning of the values even clearer
